include/dr/mhp/halo.hpp

```diff
@@ -349,12 +359,12 @@ class span_halo : public span_halo_impl<T, Memory> {
     std::vector<group_type> owned;
     drlog.debug(nostd::source_location::current(),
                 "owned groups {}/{} first/last\n", comm.first(), comm.last());
-    if (hb.prev > 0 && (hb.periodic || !comm.first())) {
-      owned.emplace_back(span.subspan(hb.prev, hb.prev), comm.prev(),
+    if (hb.next > 0 && (hb.periodic || !comm.first())) {
+      owned.emplace_back(span.subspan(hb.next, hb.next), comm.prev(),
                          communicator::tag::halo_reverse);
     }
-    if (hb.next > 0 && (hb.periodic || !comm.last())) {
-      owned.emplace_back(span.subspan(rng::size(span) - 2 * hb.next, hb.next),
+    if (hb.prev > 0 && (hb.periodic || !comm.last())) {
```

Comment:
Please don't do release this change in this PR. Please finish and merge https://github.com/oneapi-src/distributed-ranges/pull/324 PR first